src/main/java/qna/domain/Question.java

```diff
@@ -94,4 +87,45 @@ public void setWriter(User writer) {
         this.writer = writer;
         this.writer.addQuestion(this);
     }
+
+    public void delete(User loginUser) throws CannotDeleteException {
+        verifyDeletable(loginUser);
+        deleteAnswers(filterDeleteFalse(),loginUser);
+        createHistory(loginUser);
+    }
+
+    private void verifyDeletable(User loginUser) throws CannotDeleteException {
+        if (!isOwner(loginUser)) {
+            throw new CannotDeleteException("질문을 삭제할 권한이 없습니다.");
+        }
+        deleted = true;
+    }
+
+    private DeleteHistory createHistory(User loginUser) {
+        return DeleteHistory.addHistory(ContentType.QUESTION,id,loginUser);
+    }
+
+    private void deleteAnswers(List<Answer> answers, User loginUser) throws CannotDeleteException {
+        for( Answer answer : answers) {
+            answer.delete(loginUser);
+        }
+    }
+
+    private List<Answer> filterDeleteFalse() {
+        List<Answer> filteredAnswer = new ArrayList<>();
+        for ( Answer answer: getAnswers()) {
+            addAnswerIfNotDeleted(filteredAnswer, answer);
+        }
+        return filteredAnswer;
```

Comment:
```suggestion
        return answers.stream()
                .filter(a -> !a.isDeleted())
                .collect(Collectors.toList());
```
위와 같이 람다식으로 표현해 볼 수 있을 것 같습니다 😄 
그리고 이 부분도 `Answer`와 관련된 기능이므로 `Answers` 위임 되면 좋을 것 같습니다 😄